### DISCUSSION TITLE
Why can't remote_ts be configured as 0.0.0.0/0

### DISCUSSION THREAD

**Discussion by XYQ202** [strongswan/strongswan#1172] [Q&A]
Hello,Team
          There is a problem with the configuration of ipsec. I haven't found a good method. Can you give some advice?

My swanctl.conf file is configured as follows:
`connections {

OptusIPSEC {
      local_addrs = 10.201.189.34
      remote_addrs = 192.168.201.33

      local {
         auth = pubkey
         certs = CrossfireCert.pem
         id= @crossfire.org
      }
      remote {
         auth = pubkey
         id = @gateway.org
      }
      children {
OptusIPSEC {
            local_ts = 10.201.189.104/32
            remote_ts = 0.0.0.0/0

            updown = /usr/libexec/ipsec/_updown iptables
            rekey_time = 7200
            rekey_bytes = 500000000
            rekey_packets = 1000000
            dpd_action = restart
            esp_proposals= aes256-sha256
         }
      }
      version = 2
      mobike = no
      reauth_time = 10800
      dpd_delay = 15
      unique = no
      proposals = aes256-sha256-modp1024
   }
}`

When I configure **remote_ts** to **0.0.0.0/0**, does it mean that any network segment will be encrypted? When the tunnel is established, I can't SSH into the inside of the device.
When I set **remote_ts** to something else (like **0.0.0.0/8**), the tunnel can be established successfully, and I can SSH into the inside of the device.

Why does this happen? Is there any way to change it to 0.0.0.0/0, and I can still log in to the device through SSH?

**Comment by XYQ202** [comment]
```
connections {

OptusIPSEC {
      local_addrs = 10.201.189.34
      remote_addrs = 192.168.201.33

      local {
         auth = pubkey
         certs = CrossfireCert.pem
         id= @crossfire.org
      }
      remote {
         auth = pubkey
         id = @gateway.org
      }
      children {
OptusIPSEC {
            local_ts = 10.201.189.104/32
            remote_ts = 0.0.0.0/0

            updown = /usr/libexec/ipsec/_updown iptables
            rekey_time = 7200
            rekey_bytes = 500000000
            rekey_packets = 1000000
            dpd_action = restart
            esp_proposals= aes256-sha256
         }
      }
      version = 2
      mobike = no
      reauth_time = 10800
      dpd_delay = 15
      unique = no
      proposals = aes256-sha256-modp1024
   }
}
```

**Comment by tobiasbrunner** [comment]
> When I configure remote_ts to 0.0.0.0/0, does it mean that any network segment will be encrypted?

Obviously.

> Is there any way to change it to 0.0.0.0/0, and I can still log in to the device through SSH?

Sure, install a bypass policy for SSH.

**Reply by XYQ202** [reply]
How to install bypass policy for SSH？
Can you provide more details? 
0.0.0.0/0 means that all network segments will be encrypted, and as far as I know, the SSH process in the device should also be encrypted, so I can't access the device through SSH. I don't know how to configure.

**Reply by tobiasbrunner** [reply]
See [here](https://docs.strongswan.org/docs/5.9/config/quickstart.html#_passthroughbypass_policies).

**Reply by XYQ202** [reply]
> See [here](https://docs.strongswan.org/docs/5.9/config/quickstart.html#_passthroughbypass_policies).

What I want to ask is that in my **swanctl.conf** configuration file above, in addition to the connections configuration of OptusIPSEC, can I also add the configuration of passthrough-subnet to the connections section? This means that there are two configuration files in the connections section, as follows:

```
connections {
      passthrough-subnet {
          remote_addrs = 127.0.0.1
      children {
        passthrough-subnet {
          local_ts  = 10.1.0.0/24
          remote_ts = 192.168.2.0/24
          mode = pass
          start_action = trap
        }
      }
    }
      OptusIPSEC {
      local_addrs = 10.201.189.34
      remote_addrs = 192.168.201.33

      local {
         auth = pubkey
         certs = CrossfireCert.pem
         id= @crossfire.org
      }
      remote {
         auth = pubkey
         id = @gateway.org
      }
      children {
         OptusIPSEC {
            local_ts = 10.201.189.104/32
            remote_ts = 0.0.0.0/0

            updown = /usr/libexec/ipsec/_updown iptables
            rekey_time = 7200
            rekey_bytes = 500000000
            rekey_packets = 1000000
            dpd_action = restart
            esp_proposals= aes256-sha256
         }
      }
      version = 2
      mobike = no
      reauth_time = 10800
      dpd_delay = 15
      unique = no
      proposals = aes256-sha256-modp1024
   }
}
```

**Reply by tobiasbrunner** [reply]
Yes, you can have as many sub-sections in `connections` as you want.
